### PR TITLE
feat: Add prefetching to all home feed modules

### PR DESCRIPTION
**Pull request by olerichter00**
The type of this PR is: **Feature**

This PR resolves [CX-1780]

### Description

 Add prefetching to all home feed modules.

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/main/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/main/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [ ] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Add prefetching to all home screen modules - ole

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[CX-1780]: https://artsyproduct.atlassian.net/browse/CX-1780?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against cbaf22f4403882d899059b90f2daf074aebc138c
</p>

File: src/lib/Scenes/Articles/Articles.tsx [hunk 1]
```diff
@@ -49,12 +49,14 @@ export const Articles: React.FC<ArticlesProps> = (props) => {
   )
 }
 
+export const articlesDefaultVariables = {
```
Comment (olerichter00) [inline src/lib/Scenes/Articles/Articles.tsx:52]: exporting the default variables here to use them to prefetch the query.

File: src/lib/Scenes/Home/Home.tsx [hunk 2]
```diff
@@ -185,24 +206,8 @@ const Home = (props: Props) => {
...
-          prefetchVariablesExtractor={(item) => {
-            switch (item?.type) {
-              case "lotsByFollowedArtists":
-                return lotsByArtistsYouFollowDefaultVariables()
-            }
-
-            return {}
-          }}
+          prefetchUrlExtractor={(item) => item?.prefetchUrl}
+          prefetchVariablesExtractor={(item) => item?.prefetchVariables}
```
Comment (olerichter00) [inline src/lib/Scenes/Home/Home.tsx:210]: By moving the `prefetchUrl` and the `prefetchVariables` into the modules list we can simplify the logic.

Comment (MounirDhahri) [reply src/lib/Scenes/Home/Home.tsx:210]: nice

**Review by MounirDhahri — APPROVED** [review]
Nice thanks Ole

**Pull request merged**